foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigUtil.java

```diff
@@ -161,31 +162,42 @@ public static AbstractConfiguration convertEnvVariable(AbstractConfiguration sou
     return source;
   }
 
-  //inject a copy of servicecomb.xxx for cse.xxx
+  //主要是为了兼容老的版本
   private static void duplicateCseConfigToServicecomb(AbstractConfiguration source) {
-    Iterator<String> keys = source.getKeys();
-    while (keys.hasNext()) {
-      String key = keys.next();
-      if (!key.startsWith(CONFIG_CSE_PREFIX)) {
-        continue;
-      }
+    duplicateCseConfigToServicecombWithSourceName(source, DEFAULT_SOURCE_NAME);
+  }
+
 
-      String servicecombKey = CONFIG_SERVICECOMB_PREFIX + key.substring(key.indexOf(".") + 1);
-      if (!source.containsKey(servicecombKey)) {
-        source.addProperty(servicecombKey, source.getProperty(key));
-      } else {
-        LOGGER
-            .warn(
-                "Key {} with an ambiguous item {} exists, it's recommended to use only one of them.",
-                key, servicecombKey);
+  //inject a copy of servicecomb.xxx for cse.xxx
+  private static void duplicateCseConfigToServicecombWithSourceName(AbstractConfiguration source,String sourceName) {
```

Comment:
This parameter is better to use boolean to indicate if need to duplicate (yaml file has already duplicated, so no need to duplicate)